qiita_db/software.py

```diff
@@ -0,0 +1,533 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from .base import QiitaObject
+from .sql_connection import TRN
+from .exceptions import (QiitaDBDuplicateError, QiitaDBError,
+                         QiitaDBUnknownIDError)
+from .util import get_table_cols, get_table_cols_w_type
+
+
+class Command(QiitaObject):
+    r"""An executable command available in the system
+
+    Attributes
+    ----------
+    name
+    description
+    cli
+    parameters_table
+
+    Methods
+    -------
+    create
+
+    See Also
+    --------
+    qiita_db.software.Software
+    """
+    _table = "software_command"
+
+    def _check_id(self, id_):
+        """Check that the provided ID actually exists in the database
+
+        Parameters
+        ----------
+        id_ : int
+            The ID to test
+
+        Notes
+        -----
+        This function overwrites the base function, as the sql layout doesn't
+        follow the same conventions done in the other classes.
+        """
+        with TRN:
+            sql = """SELECT EXISTS(
+                        SELECT *
+                        FROM qiita.software_command
+                        WHERE command_id = %s)"""
+            TRN.add(sql, [id_])
+            return TRN.execute_fetchlast()
+
+    @classmethod
+    def create(cls, software, name, description, cli_cmd, parameters_table):
+        r"""Creates a new command in the system
```

Comment:
Maybe the tuple (software, cli_cmd)?

As an example using QIIME: split_librares.py. The parameters table did not change between 1.8.0 and 1.9.0, so we can re-use the parameters_table. However, the software would be different, since one will be the 1.8.0 version and the other one would be 1.9.0.

Does it makes sense?